bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,172 @@
+    search_range = [(calc_line_col(file, start), calc_line_col(file, end))
+                    for (start, end) in search_range]
+
+    for i in search_range:
+        yield SourceRange.from_values(filename,
+                                      start_line=i[0][0],
+                                      start_column=i[0][1],
+                                      end_line=i[1][0],
+                                      end_column=i[1][1])
+
+
+def find_string_comment(file, annot, escape):
+    """
+    gives all instances of strings or multiline comments found within
+    the file, even if they are nested in other strings or comments
+    """
+    if escape:
+        search_func = unescaped_search_in_between
+    else:
+        search_func = search_in_between
+    found_pos = ()
+    for annot_type in annot:
+        found_pos += tuple(search_func(annot_type, annot[annot_type], file))
+    if found_pos:
+        found_pos = tuple((i.begin.range[0], i.end.range[1])
+                          for i in found_pos)
+    return found_pos
+
+
+def find_singleline_comments(file, annot, match_pos):
+    """
+    Finds all single-line comments outside of other string or comments.
+    """
+    single_comm = []
+    for comment_type in annot:
+        for i in search_for(comment_type, file):
+            if not in_range_list(match_pos, (i.start(), i.end())):
+                end = file.find('\n', i.start())
+                if end != -1:
+                    single_comm.append((i.start(), end))
+                else:
+                    end = len(file) + 1
+                    single_comm.append((i.start(), end))
+    return tuple(single_comm)
+
+
+def in_range_list(outside_range_list, inside_range):
+    """
+    finds if a given 'range' is inside a any of the 'ranges' in
+    a list of ranges.
+    """
+    for outside_range in outside_range_list:
+        if inside_range == outside_range:
+            continue
+
+        elif inside_range[0] == outside_range[0]:
+            if inside_range[1] < outside_range[1]:
+                return True
+
+        elif inside_range[0] in range(outside_range[0], outside_range[1]):
+            return True
+    return False
+
+
+def calc_line_col(file, pos_to_find):
+    """
+    Calculate line number and column in the file, from position.
+    """
+    line = 1
+    pos = -1
+    pos_new_line = file.find('\n')
+    while True:
+        if pos_new_line == -1:
+            return (line, pos_to_find-pos)
+
+        if pos_to_find <= pos_new_line:
+            return (line, pos_to_find - pos)
+
+        else:
+            line += 1
+            pos = pos_new_line
+            pos_new_line = file.find('\n', pos_new_line + 1)
+
+
+def remove_nested(match_pos):
+    """
+    removes all the entries from a listthat are nested
+    inside some other entry of that list.
+    """
+    unnested_list = []
+    for i in match_pos:
+        if not in_range_list(match_pos, i):
+            unnested_list.append(i)
```

Comment:
`tuple(filter(lambda x: in_range_list(match_pos, x), match_pos))` ?
Does this need a new function ?
It's only used once